04_documentation/code_documentation_tools_demo.md

```diff
@@ -0,0 +1,302 @@
+
+- Change back to the root of the repository
+- Reformat the `README.md` into `README.rst`
+
+    1. Rename the file
+
+       ```bash
+       git mv README.md README.rst
+       ```
+
+       We do this also for consistency with the new files that will be created.
+    2. Redo headlines by underlining them
+    3. Redo inline code samples with double backticks
+    4. Reformat code block using `code-block` directive
+- Set up documentation
+    1. If not already installed, we install Sphinx via pip
+
+       ```bash
+       pip install -U sphinx
+       ```
+
+    2. Create documentation directory
+
+       ```bash
+       mkdir docs && cd docs
+       ```
+
+    3. Create basic configuration
+
+       ```bash
+       sphinx-quickstart
+       ```
+
+       Example output:
+
+       ```bash
+       Welcome to the Sphinx 4.3.1 quickstart utility.
+
+       Please enter values for the following settings (just press Enter to
+       accept a default value, if one is given in brackets).
+
+       Selected root path: .
+
+       You have two options for placing the build directory for Sphinx output.
+       Either, you use a directory "_build" within the root path, or you separate
+       "source" and "build" directories within the root path.
+       > Separate source and build directories (y/n) [n]: n
+
+       The project name will occur in several places in the built documentation.
+       > Project name: Diffusion Solver
+       > Author name(s): Alexander Jaust
+       > Project release []: 0.1
+
+       If the documents are to be written in a language other than English,
+       you can select a language here by its language code. Sphinx will then
+       translate text that it generates into that language.
+
+       For a list of supported codes, see
+       https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-language.
+       > Project language [en]: en
+
+       Creating file /home/jaustar/teaching/courses/simulation-software-engineering/preparation/documentation/python-code-documentation/docs/conf.py.
+       Creating file /home/jaustar/teaching/courses/simulation-software-engineering/preparation/documentation/python-code-documentation/docs/index.rst.
+       Creating file /home/jaustar/teaching/courses/simulation-software-engineering/preparation/documentation/python-code-documentation/docs/Makefile.
+       Creating file /home/jaustar/teaching/courses/simulation-software-engineering/preparation/documentation/python-code-documentation/docs/make.bat.
+
+       Finished: An initial directory structure has been created.
+
+       You should now populate your master file /home/jaustar/teaching/courses/simulation-software-engineering/preparation/documentation/python-code-documentation/docs/index.rst and create other documentation
+       source files. Use the Makefile to build the docs, like so:
+          make builder
+       where "builder" is one of the supported builders, e.g. html, latex or linkcheck.
+       ```
```

Comment:
Sth you do here, md doesn't like?

Reply:
I am not sure. For me it looks good on GitHub and when using pandoc.